repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -168,6 +173,53 @@ def _import_gpg_keys(context, install_root_dir, target_major_version):
         )
 
 
+def _handle_transaction_err_msg_size_old(err):
+    # NOTE(pstodulk): This is going to be removed in future!
+
+    article_section = 'Generic case'
+    xfs_info = next(api.consume(XFSPresence), XFSPresence())
+    if xfs_info.present and xfs_info.without_ftype:
+        article_section = 'XFS ftype=0 case'
+
+    message = ('There is not enough space on the file system hosting /var/lib/leapp directory '
+               'to extract the packages.')
+    details = {'hint': "Please follow the instructions in the '{}' section of the article at: "
+                       "link: https://access.redhat.com/solutions/5057391".format(article_section)}
+
+    raise StopActorExecutionError(message=message, details=details)
+
+
+def _handle_transaction_err_msg_size(err):
+    if get_env('LEAPP_OVL_LEGACY', '0') == '1':
+        _handle_transaction_err_msg_old(err)
+        return  # not needed actually as the above function raises error, but for visibility
+    NO_SPACE_STR = 'more space needed on the'
+
+    # Disk Requirements:
+    #   At least <size> more space needed on the <path> filesystem.
+    #
+    missing_space = [line.strip() for line in err.stderr.split('\n') if  NO_SPACE_STR in line]
+    size_str = re.match(r'At least (.*) more space needed', missing_space[0]).group(1)
+    message = 'There is not enough space on the file system hosting /var/lib/leapp.'
+    hint = (
+        'Increase the free space on the filesystem hosting'
+        ' /var/lib/leapp by {} at minimum. It is suggested to provide'
+        ' reasonably more space to be able to perform all planned actions'
+        ' (e.g. when 200MB is missing, add 1700MB or more).\n\n'
+        'It is also a good practice to create dedicated partition for'
+        ' for /var/lib/leapp when more space is needed, which can be'
+        ' dropped after the system upgrade is fully completed'
+        ' For more info, see: {}'
+        .format(size_str, DEDICATED_URL)
+    )
+    # we do not want to confuse customers by the orig msg speaking about
+    # missing space on '/'. Skip the Disk Requirements section.
+    # The information is part of the hint.
+    details = {'hint': hint}
+
+    raise StopActorExecutionError(message=message, details=details)
```

Comment:
I am not sure whether this is actually correct. I am afraid that I maybe missed a corner case here. Imagine this partitioning:
```
# mount point: free space
/: 10 GB
/var: 700MB (e.g.)
/var/lib/leapp: 10 GB
```
In this case, the disk img for /var is located under the leapp dir, but it's apparent size will be only ~ 665MB. Currently we need to download ~160M data (metadata + packages). And some space is consumed running dnf/rpm. This all is stored inside the OVL layer.  Imagine we need to download more pkgs (/var cannot have less than 500M otherwise it crashes during the scan phase when we call yum :)), in this only one situation, I think we will have problem with the msg, as it will be actually affected by the size of /var and not /var/lib/leapp. But who would actually try to upgrade when having only such small space on /var ??? So it's from my POV really total corner case that should not happen. Also it needs to be differenciated, whether the problem is caused during pkgs download or calculation whether there is enough disk space for the installation of the downloaded packages (both cases has different answer where the space is missing).

Opposite problem could appear if /var/lib/leapp is not big enough and we consume for the pkg download all remaining free space. But regarding the implemented checks, I do not believe this situation will happen (it would mean that someone would like to download couple of GBs pkgs to install super big container right now. and that does not seem meaningful.

Reply:
@oamg/developers  ^ ping. do not miss this comment

Reply:
The only scenario that I can image this to happen is that someone has small /var. The first leapp execution crashes, lacking required disk space. The user then deals with this problem by creating a separate /var/lib/leapp partition without increasing the size of /var. The wording of this error message should not create the impression that a separate /var/lib/leapp will not deal with the case when the /var partition is too small.

I've also checked the internet for recommended partitioning. [RHEL8 partitioning guide](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/performing_an_advanced_rhel_8_installation/partitioning-reference_installing-rhel-as-an-experienced-user) recommends that `5gb` is enough for `/var`. [RHEL7 guide](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/installation_guide/sect-disk-partitioning-setup-x86#sect-recommended-partitioning-scheme-x86) does not mention a precise number, but says it should be "sufficiently large", whatever that means. Other advice on the internet states that `/var` should be at least 2gb in size.

I've also went trough vanilla RHEL images on Azure and they all have `/var` 5gb+ in size (I remember encountering some issues around /var space there before).

Therefore, I do not think that this corner case will be encountered.